control-plane-operator/controllers/hostedcontrolplane/csi/kubevirt/kubevirt.go

```diff
@@ -0,0 +1,386 @@
+package kubevirt
+
+import (
+	"bytes"
+	"context"
+	"embed"
+	"fmt"
+	"io"
+
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/manifests"
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/pki"
+	"github.com/openshift/hypershift/support/upsert"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//go:embed files/*
```

Comment:
Can we either put this into support/ as in https://github.com/openshift/hypershift/pull/1698/files#diff-091a3abbf9e278c6f5cd74ca5fb7094a370f2a57c8c13c3f6568a6cc049ed8ea

Also why particular reason why using yaml here vs golang?

Reply:
These yamls are a slightly modified version of the yamls sourced from the deploy yaml in the kubevirt-csi repo. I'll be modifying the kubevirt-csi repo yamls to match what we had to do for hypershift, so eventually they should be a 1:1 reflection of one another. The idea would be that when we test kubevirt-csi upstream, we'd be using the same yamls that get sourced into hypershift... but we're not 100% there yet.

In the medium term, we're looking to converge with the CSO which will involve sourcing yaml similar to what we're doing here.

Reply:
If we use yamls we the must* functions already exists in olm/assets, please move them to support/assets and reuse them.

However I don't think we can use yaml to render stuff management side because we set dynamic topology/affinity opinions over these components, see `SetDefaults()`